### DISCUSSION TITLE
[BUG] Editing frequency of a subscription creates a new order and charges customer immediately

### DISCUSSION THREAD

**Discussion by Krislunde** [swellstores/community#439] [Help] [answered]
In the past, there used to be a setting in the swell admin panel settings for whether to allow customers to edit the frequency and options  of an active subscription - which was fetched through `settings.account.subscriptions.allowFrequencyEdit` and `settings.account.subscriptions.allowOptionsEdit` both of these now seem to be gone?

On our frontend we had functionality for our clients to edit their subscriptions with the above, which was executed with `await swell.subscriptions.update(subscription.id, options)` - However, while this does indeed update the subscription, it also immediately creates an order to be fulfilled immediately and charges the customer for it. Has something changed in the backend? What's the deal here? This seems like a pretty serious bug - we've had to remove the ability of our customers to edit their subscriptions for now until hearing from you.

Thank you.

**Comment by ericingram** [comment]
I did a quick search and found those settings are part of Origin theme, and not in the backend

**Reply by Krislunde** [reply]
Alright, fair enough. However, it doesn't seem right that a new order should be placed and immediately billed when you try to update a subscription with new frequency does it? Such as changing delivery from every month to every two months. It correctly updates the subscription frequency, but also immediately generates a new order.

Edit: the implementation is taken from the Origin theme.

**Reply by ericingram** [reply]
That doesn’t sound right at face value

cc @logeshswell

**Reply by logeshswell** [reply]
@Krislunde Can you share a sample subscription URL, that generated orders?
Also, let me know the subscription frequency before and after the update?

**Reply by Krislunde** [reply]
Hi @logeshswell ! Happy to help, but I'm not sure what URL you're looking for. We have a separate storefront. 

I've attached [some images](https://imgur.com/a/bTIL6Ot) to help visualise the issue- the code implementation is taken from the Origin storefront. Let me know what I can do to assist further.

**Reply by logeshswell** [reply, marked as answer]
Thanks for the images, it really helped me understand the problem better.

When do you wish the subscription should generate an order if the frequency is changed now?

Also, we don't support scheduled frequency corrections for the subscription at the moment, but you can achieve it using Swell apps which will be launched soon.

Subscription will generate orders (as much as required) and capture the charge when the frequency is changed. Otherwise, you can schedule a cron to update the subscription frequency at the desired time.

**Reply by Krislunde** [reply]
Well, the scenario is this: 

Customer has a subscription with a delivery every month. Subscription is set to renew on the 13th of December. Customer changes the frequency to be `Every 2 Months` and now the subscription say it is set to renew on the 13th of January - however, Swell still generates a new order immediately.

The expected scenario is for the subscription to only generate a new order when the next renewal date starts. Is this not the case?

**Reply by logeshswell** [reply]
Yeah, it is not supported at the moment. This is what I call as scheduled frequency update. 
If you want to generate a new order on the next renewal date, then the workaround is to update the frequency on the day of next renewal date.

**Reply by shc023** [reply]
@logeshswell that doesn't sound quite right as a workaround either. Continuing the example @Krislunde used above, if the subscription is currently `Dec 13` and the user wants to change from `every 1 month` to `every 2 months`, that means the user doesn't want the subscription to fulfill on `Dec 13`, only `Jan 13`. 

If we're just waiting until `Jan 13` `to update the frequency on the day of next renewal date.`, the order would still be created on `Dec 13`, which would be the incorrect behavior.

**Reply by Krislunde** [reply]
To add on, nothing in the origin storefront indicates the customer will be charged immediately for an order either — it merely makes the customer believe they can choose the frequency at will.

I understand that Swell needs to cancel the current subscription and create a new one in the backend — but it should just cancel current subscription, and set up a new one to be fulfilled *for the first time* on the new date, not immediately charge the customer and create an order.

Can't say I'm thrilled about this being marked as "answered".

**Reply by shc023** [reply]
maybe @sags95 might know a better workaround for this, having answered a related question I just had
